File: support/pg/src/main/java/com/coffee/gu/toss/Payment.java
package com.coffee.gu.toss;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Payment(
        String paymentKey,
        String orderId,
        TossPaymentsStatus status,
        BigDecimal totalAmount,
        String method,
        Card card,
        EasyPay easyPay,
        OffsetDateTime approvedAt,
        List<Cancel> cancels
) {
    public record Card(
            String approveNo
    ) {}
    public record EasyPay(
            String provider,
            BigDecimal amount,
            BigDecimal discountAmount
    ) {}
    public record Cancel(
            BigDecimal cancelAmount,
            String cancelReason,
            BigDecimal taxFreeAmount,
            BigDecimal taxExemptionAmount,
            BigDecimal refundableAmount,
            BigDecimal cardDiscountAmount,
            BigDecimal transferDiscountAmount,
            BigDecimal easyPayDiscountAmount,
            OffsetDateTime canceledAt,
            String transactionKey,
            String receiptKey,
            String cancelStatus
    ) {}
}

File: support/pg/src/main/kotlin/com/coffee/gu/toss/Payment.kt
package com.coffee.gu.toss

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.math.BigDecimal
import java.time.OffsetDateTime

@JsonIgnoreProperties(ignoreUnknown = true)
class Payment(
    val paymentKey: String,
    val orderId: String,
    val status: TossPaymentsStatus,
    val totalAmount: BigDecimal,
    val method: String? = null,
    val card: Card? = null,
    val easyPay: EasyPay? = null,
    val approvedAt: OffsetDateTime? = null,
    val cancels: List<Cancel>? = null
) {
    class Card(
        val approveNo: String? = null
    )

    class EasyPay(
        val provider: String? = null,
        val amount: BigDecimal? = null,
        val discountAmount: BigDecimal? = null
    )

    class Cancel(
        val cancelAmount: BigDecimal? = null,
        val cancelReason: String? = null,
        val taxFreeAmount: BigDecimal? = null,
        val taxExemptionAmount: BigDecimal? = null,
        val refundableAmount: BigDecimal? = null,
        val cardDiscountAmount: BigDecimal? = null,
        val transferDiscountAmount: BigDecimal? = null,
        val easyPayDiscountAmount: BigDecimal? = null,
        val canceledAt: OffsetDateTime? = null,
        val transactionKey: String? = null,
        val receiptKey: String? = null,
        val cancelStatus: String? = null
    )
}
